warehouse/query-core/src/main/java/datawave/query/attributes/TemporalGranularity.java

```
@@ -12,100 +12,101 @@
 
     @JsonCreator
-    public static UniqueGranularity of(String name) {
+    public static TemporalGranularity of(String name) {
+        name = name.toUpperCase();
         switch (name) {
             case "ALL":
-                return UniqueGranularity.ALL;
+                return TemporalGranularity.ALL;
             case "YEAR":
-                return UniqueGranularity.TRUNCATE_TEMPORAL_TO_YEAR;
+                return TemporalGranularity.TRUNCATE_TEMPORAL_TO_YEAR;
             case "MONTH":
-                return UniqueGranularity.TRUNCATE_TEMPORAL_TO_MONTH;
+                return TemporalGranularity.TRUNCATE_TEMPORAL_TO_MONTH;
             case "DAY":
-                return UniqueGranularity.TRUNCATE_TEMPORAL_TO_DAY;
+                return TemporalGranularity.TRUNCATE_TEMPORAL_TO_DAY;
             case "HOUR":
-                return UniqueGranularity.TRUNCATE_TEMPORAL_TO_HOUR;
+                return TemporalGranularity.TRUNCATE_TEMPORAL_TO_HOUR;
             case "TENTH_OF_HOUR":
-                return UniqueGranularity.TRUNCATE_TEMPORAL_TO_TENTH_OF_HOUR;
+                return TemporalGranularity.TRUNCATE_TEMPORAL_TO_TENTH_OF_HOUR;
             case "MINUTE":
-                return UniqueGranularity.TRUNCATE_TEMPORAL_TO_MINUTE;
+                return TemporalGranularity.TRUNCATE_TEMPORAL_TO_MINUTE;
             case "SECOND":
-                return UniqueGranularity.TRUNCATE_TEMPORAL_TO_SECOND;
+                return TemporalGranularity.TRUNCATE_TEMPORAL_TO_SECOND;
             case "MILLISECOND":
-                return UniqueGranularity.TRUNCATE_TEMPORAL_TO_MILLISECOND;
+                return TemporalGranularity.TRUNCATE_TEMPORAL_TO_MILLISECOND;
             default:
-                throw new IllegalArgumentException("No " + UniqueGranularity.class.getSimpleName() + " exists with the name " + name);
+                throw new IllegalArgumentException("No " + TemporalGranularity.class.getSimpleName() + " exists with the name " + name);
```

This exception gets swallowed by GroupBy.validate()

---

So it does. Fixing.